### DISCUSSION TITLE
problem with using a custom EJS template -- Quarto rendering interferes and changes the html

### DISCUSSION THREAD

**Discussion by paciorek** [quarto-dev/quarto-cli#7310] [Q&A] [answered]
### Description

I'm hoping to use my own EJS template that creates an HTML table. When I try to do so, I find that the Quarto rendering process is somehow modifying the HTML in my template, perhaps because Quarto is trying to process it as a Quarto listings table. 

Here's a minimal example. First the qmd file where I initially just create a basic HTML table and then try to create the equivalent HTML table in an EJS template (albeit without having the template fill in any values -- it's all just straight HTML for simplicity here). 

````qmd
---
title: "Test"
listing:
  - id: test
    template: test.ejs
---

A table directly.

<table>
<tbody>
<tr>
<th scope="row" style="width: 20%">Hello</th>
<td style="width: 40%;"><strong style="background-color:purple; border-radius: 9px; padding: 5px;">text</strong></td>
<td style="width: 40%;">2</td>
</tr>
</tbody>
</table>

A table via a template.

::: {#test}
:::
````

Here is the EJS file.

````
```{=html}
<table>
<tbody>
<tr>
<th scope="row">Hello</th>
<td><strong style="background-color:purple; border-radius: 9px; padding: 5px;">text</strong></td>
<td>1</td>
</tr>
</tbody>
</table>
```
````

The result in the rendered .html file is that the HTML from the template has been modified. The `th` element has been turned into a `td` element and the styling on the `strong` element has been removed.  Here's the relevant HTML produced for the template version of the table.

```{=html}
<table data-quarto-postprocess="true" class="table">
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th" scope="row">Hello</td>
<td><strong>text</strong></td>
<td>1</td>
</tr>
</tbody>
</table>
```

Is there any way to just have the EJS template processed without Quarto doing additional processing? 

Side note: I've found that I can hack it in some cases (e.g., by inserting a dummy `<b></b>` into the template before `</tbody<`, which causes this warning message ` WARNING: Unable to parse table from raw html block: skipping.` when rendering. The result of that 'skipping' is that the HTML in my template is preserved (i.e., the `th` and `strong` are not messed with).

**Comment by mcanouil** [comment]
Could you format your post properly?
Also raw HTML should be in raw block to avoid Pandoc/Quarto parsing it weirdly, *i.e.*, `` ```{=html} ``

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by paciorek** [reply]
Whoops, I fixed the missing backtick and put the html in a raw block. Thanks/sorry.

**Reply by cscheid** [reply, marked as answer]
Take a look at this https://quarto.org/docs/authoring/tables.html#disabling-quarto-table-processing

**Comment by paciorek** [comment]
Thanks, that worked.

Is there something that can be done in the Quarto processing system to better handle this sort of situation -- namely that when a custom EJS template is used, it is left alone? It's not clear to me if the behavior I was experiencing is what the developers intend?
